inception-log/src/main/java/de/tudarmstadt/ukp/inception/log/EventRepositoryImpl.java

```diff
@@ -63,24 +63,88 @@ public void create(LoggedEvent... aEvents)
 
     @Override
     @Transactional
-    public List<LoggedEvent> listLoggedEvents(Project aProject, String aUsername, String aEventType,
-            int aSize, long recommenderId)
+    public List<LoggedEvent> listLoggedEventsForRecommender(Project aProject, String aUsername,
+            String aEventType, int aMaxSize, long aRecommenderId)
     {
-        String query = String.join("\n",
-                "FROM LoggedEvent WHERE ",
+        String detailStr = "%\"recommenderId\":" + aRecommenderId + "%";
+
+        return listLoggedEventsForDetail(aProject, aUsername, aEventType, aMaxSize, detailStr);
+    }
+    
+    @Override
+    @Transactional
+    public List<LoggedEvent> listLoggedEventsForDetail(Project aProject, String aUsername,
+            String aEventType, int aMaxSize, String aDetail)
+    {
+        String query = String.join("\n", 
+                "FROM LoggedEvent WHERE ", 
                 "user=:user AND ",
-                "project = :project AND ",
-                "event = :event AND ",
+                "project = :project AND ", 
+                "event = :event AND ", 
                 "details LIKE :details ",
                 "ORDER BY created DESC");
 
         return entityManager.createQuery(query, LoggedEvent.class)
                 .setParameter("user", aUsername)
                 .setParameter("project", aProject.getId())
                 .setParameter("event", aEventType)
-                .setParameter("details", "%\"recommenderId\":" + recommenderId + "%")
-                .setMaxResults(aSize)
-                .getResultList();
+                .setParameter("details", aDetail)
+                .setMaxResults(aMaxSize).getResultList();
+    }
+    
+    @Override
+    @Transactional
+    public List<LoggedEvent> listUniqueLoggedEventsForDoc(Project aProject, String aUsername,
+            String[] aEventTypes, int aMaxSize)
+    {
+        String eventsQuery = buildEventsCondition(aEventTypes.length);
+        String query = String.join("\n", 
+                "FROM LoggedEvent WHERE",
+                "id IN",
+                // select one event when time-stamps are the same per document
+                "   (SELECT max(id)",
+                "   FROM LoggedEvent WHERE",
+                "   user=:user AND",
+                "   project=:project AND",
+                    eventsQuery,
```

Comment:
why not use a "IN" here as well instead of string-concatenating the event conditions?

Reply:
Example:

```
            Query q = entityManager.createQuery(
                    "UPDATE AnnotationFeature \n" +
                    "SET type = :fixedType \n" +
                    "WHERE type = :oldType \n" +
                    "AND name in (:featureNames)");            
            q.setParameter("featureNames",
                    Arrays.asList(COREFERENCE_RELATION_FEATURE, COREFERENCE_TYPE_FEATURE));

```

Reply:
Thanks, I wasn't aware that it is possible to add params as lists.